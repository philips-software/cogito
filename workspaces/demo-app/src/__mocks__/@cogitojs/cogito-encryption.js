export class CogitoKeyProvider {
  async createNewKeyPair () {
    return createNewKeyPairMock()
  }

  async getPublicKey ({ tag }) {
    return getPublicKeyMock({ tag })
  }
}

export class CogitoEncryption {
  async encrypt ({ jsonWebKey, plainText }) {
    return encryptMock({ jsonWebKey, plainText })
  }
}

export const createNewKeyPairMock = jest.fn()
export const getPublicKeyMock = jest.fn()
export const encryptMock = jest.fn()
