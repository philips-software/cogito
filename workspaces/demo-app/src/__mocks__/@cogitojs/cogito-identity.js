export class CogitoIdentity {
  static Property = {
    Username: 'username',
    EthereumAddress: 'ethereumAddress'
  }
  constructor ({ channel }) {
    this.channel = channel
  }
  getInfo () {
    if (this.channel.error) {
      return Promise.reject(this.channel.error)
    } else {
      return Promise.resolve(this.channel.mockIdentityInfo())
    }
  }
}
