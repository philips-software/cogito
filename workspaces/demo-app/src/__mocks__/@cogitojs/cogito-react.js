import React from 'react'

export class CogitoReact extends React.Component {
  state = {
    cogitoWeb3: {},
    telepathChannel: {
      id: 'telepathId',
      key: Uint8Array.from([1, 2, 3]),
      appName: 'Cogito Demo App'
    },
    contractsProxies: {
      SimpleStorage: 'SimpleStorageProxy'
    },
    newChannel: jest.fn()
  }

  render () {
    const { render, children } = this.props
    return render ? render(this.state) : children(this.state)
  }
}
