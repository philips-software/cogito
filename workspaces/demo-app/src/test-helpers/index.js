export { inRouter, renderInRouter } from './router'
export { EventWaiter } from './EventWaiter'
export { GanacheTestNetwork } from './GanacheTestNetwork'
