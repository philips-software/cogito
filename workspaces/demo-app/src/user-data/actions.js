class UserDataActions {
  static connectionEstablished = () => ({
    type: 'CONNECTION_ESTABLISHED'
  })

  static clearConnectionEstablished = () => ({
    type: 'CLEAR_CONNECTION_ESTABLISHED'
  })

  static setAccount = account => ({
    type: 'SET_USER_ACCOUNT',
    account
  })

  static setTelepath = ({ channelId, channelKey, appName }) => ({
    type: 'SET_TELEPATH',
    channelId,
    channelKey,
    appName
  })

  static setBalance = balance => ({
    type: 'SET_BALANCE',
    balance
  })

  static setIdentityInfo = info => ({
    type: 'SET_IDENTITY_INFO',
    info
  })
}

export { UserDataActions }
