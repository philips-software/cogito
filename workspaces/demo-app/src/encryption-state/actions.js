import { CogitoEncryption, CogitoKeyProvider } from '@cogitojs/cogito-encryption'

class EncryptionActions {
  static setPlainText = (plainText) => ({
    type: 'ENCRYPTION_SET_PLAINTEXT',
    plainText
  })

  static setCipherText = (cipherText) => ({
    type: 'ENCRYPTION_SET_CIPHERTEXT',
    cipherText
  })

  static setKeyTag = (tag) => ({
    type: 'ENCRYPTION_SET_KEY_TAG',
    tag
  })

  static encryptionError = (message) => ({
    type: 'ENCRYPTION_ERROR',
    message
  })

  static encryptPending = () => ({
    type: 'ENCRYPTION_PENDING'
  })

  static decryptPending = () => ({
    type: 'DECRYPTION_PENDING'
  })

  static encrypt = ({ telepathChannel }) => {
    const cogitoEncryption = new CogitoEncryption({ telepathChannel })
    const cogitoKeyProvider = new CogitoKeyProvider({ telepathChannel })

    return async (dispatch, getState) => {
      dispatch(EncryptionActions.encryptPending())

      try {
        const plainText = getState().encryption.plainText
        const tag = await cogitoKeyProvider.createNewKeyPair()
        dispatch(EncryptionActions.setKeyTag(tag))

        const jsonWebKey = await cogitoKeyProvider.getPublicKey({ tag })
        const cipherText = await cogitoEncryption.encrypt({ jsonWebKey, plainText })

        dispatch(EncryptionActions.setPlainText(''))
        dispatch(EncryptionActions.setCipherText(cipherText))
      } catch (error) {
        dispatch(EncryptionActions.encryptionError(error.message))
      }
    }
  }

  static decrypt = ({ telepathChannel }) => {
    const encryption = new CogitoEncryption({ telepathChannel })

    return async (dispatch, getState) => {
      dispatch(EncryptionActions.decryptPending())

      try {
        const cipherText = getState().encryption.cipherText
        const tag = getState().encryption.keyTag
        const plainText = await encryption.decrypt({ tag, encryptionData: cipherText })
        dispatch(EncryptionActions.setPlainText(plainText))
      } catch (error) {
        dispatch(EncryptionActions.encryptionError(error.message))
      }
    }
  }
}

export { EncryptionActions }
