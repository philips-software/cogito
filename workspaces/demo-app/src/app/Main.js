import React from 'react'
import { Switch, Route } from 'react-router-dom'
import { PageCentered } from '@react-frontend-developer/react-layout-helpers'
import { CogitoReact } from '@cogitojs/cogito-react'
import { WithStore } from '@react-frontend-developer/react-redux-render-prop'
import { Dimmer, Loader, Segment } from 'semantic-ui-react'
import { UserDataActions } from 'user-data'

import { CogitoId } from 'pages/home'
import { Contracts } from 'pages/contracts'
import { SimpleEncryption } from 'pages/simple-encryption'
import { StreamEncryption } from 'pages/stream-encryption'
import { Attestations } from 'pages/attestations'
import { NoMatch404 } from 'pages/404'

import { SimpleStorage } from '@cogitojs/demo-app-contracts'

const contractsInfo = {
  deployedContractsInfo: [
    { contractName: 'simpleStorage', contractDefinition: SimpleStorage }
  ],
  rawContractsInfo: []
}

class Main extends React.Component {
  renderComponent = (Component, routeProps, web3Props) => (
    <Component {...web3Props} {...routeProps} />
  )

  web3IsReady = ({ web3, channel, contracts }) => {
    return web3 && channel && contracts
  }

  onTelepathChanged = ({ channelId, channelKey, appName }, dispatch) => {
    dispatch(UserDataActions.setTelepath({ channelId, channelKey, appName }))
  }

  render () {
    return (
      <WithStore
        selector={state => ({
          channelId: state.userData.channelId,
          channelKey: state.userData.channelKey
        })}
        render={({ channelId, channelKey }, dispatch) => (
          <CogitoReact
            contracts={contractsInfo}
            channelId={channelId}
            channelKey={channelKey}
            appName='Cogito Demo App'
            onTelepathChanged={telepath =>
              this.onTelepathChanged(telepath, dispatch)}
          >
            {web3Props => {
              if (this.web3IsReady(web3Props)) {
                return (
                  <Switch>
                    <Route
                      exact
                      path='/'
                      render={routeProps =>
                        this.renderComponent(CogitoId, routeProps, web3Props)}
                    />
                    <Route
                      exact
                      path='/contracts'
                      render={routeProps =>
                        this.renderComponent(Contracts, routeProps, web3Props)}
                    />
                    <Route
                      exact
                      path='/simple-encryption'
                      render={routeProps =>
                        this.renderComponent(
                          SimpleEncryption,
                          routeProps,
                          web3Props
                        )}
                    />
                    <Route
                      exact
                      path='/stream-encryption'
                      render={routeProps =>
                        this.renderComponent(
                          StreamEncryption,
                          routeProps,
                          web3Props
                        )}
                    />
                    <Route
                      exact
                      path='/attestations'
                      render={routeProps =>
                        this.renderComponent(
                          Attestations,
                          routeProps,
                          web3Props
                        )}
                    />
                    <Route component={NoMatch404} />
                  </Switch>
                )
              } else {
                return (
                  <PageCentered>
                    <div style={{ width: '300px', height: '200px' }}>
                      <Segment style={{ width: '100%', height: '100%' }}>
                        <Dimmer active inverted>
                          <Loader inverted content={<p>loading web3...</p>} />
                        </Dimmer>
                      </Segment>
                    </div>
                  </PageCentered>
                )
              }
            }}
          </CogitoReact>
        )}
      />
    )
  }
}

export { Main }
