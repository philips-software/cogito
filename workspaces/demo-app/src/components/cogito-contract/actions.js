import { AppEventsActions } from 'app-events'
import { UserDataActions } from 'user-data'
import { FaucetService } from 'faucet'

import { CogitoIdentity } from '@cogitojs/cogito-identity'

const getFaucetURL = () => {
  if (process.env.NODE_ENV === 'production') {
    return process.env.FAUCET_URL_PRODUCTION
  }

  return process.env.FAUCET_URL || 'http://localhost:3001/donate'
}

const transferFunds = async (account) => {
  const faucet = new FaucetService(getFaucetURL())
  await faucet.transferFunds(account)
}

const getAccount = async (getState, dispatch, channel, forceRefetchAddress) => {
  const { userData: { connectionEstablished, ethereumAddress } } = getState()

  if (connectionEstablished && !forceRefetchAddress) {
    return ethereumAddress
  }

  const requestedProperties = [
    CogitoIdentity.Property.EthereumAddress,
    CogitoIdentity.Property.Username
  ]
  const cogitoIdentity = new CogitoIdentity({ channel })
  const info = await cogitoIdentity.getInfo({ properties: requestedProperties })
  if (!info) throw new Error('No identity found on the mobile device!')
  dispatch(UserDataActions.setIdentityInfo(info))
  dispatch(UserDataActions.connectionEstablished())

  return info.ethereumAddress
}

class ContractActions {
  static increase = ({ increment, deployedContract: contract, channel, forceRefetchAddress = false }) => {
    return async (dispatch, getState) => {
      dispatch(AppEventsActions.telepathInProgress())
      try {
        const account = await getAccount(getState, dispatch, channel, forceRefetchAddress)
        // let's make sure the account has some funds
        transferFunds(account)
        await contract.increase(
          increment,
          { from: account }
        )
        dispatch(AppEventsActions.telepathFulfilled())
      } catch (error) {
        console.error(error)
        dispatch(AppEventsActions.telepathError({ reason: error.message }))
      }
    }
  }

  static read = ({ deployedContract: contract, channel, forceRefetchAddress = false }) => {
    return async (dispatch, getState) => {
      dispatch(AppEventsActions.telepathInProgress())
      try {
        await getAccount(getState, dispatch, channel, forceRefetchAddress)
        const response = await contract.read()
        const balance = response.toNumber()
        dispatch(UserDataActions.setBalance(balance))
        dispatch(AppEventsActions.telepathFulfilled())
      } catch (error) {
        console.error(error)
        dispatch(AppEventsActions.telepathError({ reason: error.message }))
      }
    }
  }
}

export { ContractActions }
