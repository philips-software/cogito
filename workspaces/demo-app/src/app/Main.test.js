import { Main } from './Main'
import { inRouter, EthereumForSimpleStorage } from 'test-helpers'
import {
  render,
  waitForElement,
  fireEvent,
  wait
} from 'test-helpers/render-props'
import { SimpleStorage } from '@cogitojs/demo-app-contracts'

import { UserDataActions } from 'user-data'
import { IdentityActions } from 'components/cogito-address/actions'

import { createStore, applyMiddleware } from 'redux'
import { rootReducer } from 'app-state/rootReducer'
import thunkMiddleware from 'redux-thunk'

jest.mock('@cogitojs/demo-app-contracts')
jest.mock('../services/documentation-loader')

jest.unmock('@cogitojs/cogito-ethereum-react')
jest.unmock('@cogitojs/cogito-identity')
jest.unmock('@react-frontend-developer/react-redux-render-prop')

describe('Main', function () {
  let store

  const testAddress = '0xabcd'
  const testUserName = 'Test User'
  const defaultIdentity = {
    ethereumAddress: testAddress,
    username: testUserName
  }
  const alternateIdentity = {
    ethereumAddress: testAddress.toUpperCase(),
    username: testUserName.toUpperCase()
  }

  const mockChannel = (channel, identity) => {
    channel.send = jest.fn().mockImplementation(request => {
      if (request.method === 'getIdentityInfo') {
        return Promise.resolve({
          result: identity
        })
      }
    })
  }

  const setUserIdentity = identity => {
    const read = IdentityActions.read
    IdentityActions.read = ({ channel }) => {
      mockChannel(channel, identity)
      return read({ channel })
    }
  }

  const setupStore = () => {
    store = createStore(
      rootReducer,
      undefined,
      applyMiddleware(thunkMiddleware)
    )
  }

  const showQRCode = async getByText => {
    const showQRCodeButton = await waitForElement(() =>
      getByText(/show qr code/i)
    )
    const {
      channelId: initialId,
      channelKey: initialKey
    } = store.getState().userData
    fireEvent.click(showQRCodeButton)
    return { initialId, initialKey, getByText, store }
  }

  const expectTelepathChanged = (initialId, initialKey, store) => {
    expect(store.getState().userData.channelId).not.toEqual(initialId)
    expect(store.getState().userData.channelKey).not.toEqual(initialKey)
  }

  const validateCorrectPageRendered = async (url, expectedText) => {
    const { container, getByText } = render(inRouter(Main, url))

    await waitForElement(() => getByText(expectedText))
    expect(container).toMatchSnapshot()
  }

  const validateTelepathChanged = async url => {
    const { getByText, store } = render(inRouter(Main, url))
    const { initialId, initialKey } = await showQRCode(getByText)
    await wait(() => {
      expectTelepathChanged(initialId, initialKey, store)
      expect(store.getState().userData.channelKey).not.toEqual(initialKey)
    })
  }

  const validatePageRerendered = async url => {
    const { getByText, store } = render(inRouter(Main, url))
    const { initialId, initialKey } = await showQRCode(getByText)
    await wait(() => {
      expectTelepathChanged(initialId, initialKey, store)
      expect(getByText(/scan the qr code/i)).toBeInTheDocument()
    })
  }

  beforeEach(async () => {
    console.log = jest.fn()
    setupStore()
    const ethereum = await EthereumForSimpleStorage.setup()
    SimpleStorage.mockImplementation(() => {
      return ethereum.deployedJSON
    })
    setUserIdentity(defaultIdentity)
  })

  afterEach(() => {
    console.log.mockRestore()
  })

  describe('when routing to home page', () => {
    const forInitialRendering = getByTestId => {
      return wait(() => {
        expect(getByTestId(/current-address/i)).toHaveTextContent(
          defaultIdentity.ethereumAddress
        )
        expect(getByTestId(/current-username/i)).toHaveTextContent(
          defaultIdentity.username
        )
      })
    }

    it('renders correct page', async () => {
      await validateCorrectPageRendered('/', 'Your Cogito account address is:')
    })

    it('creates new telepath channel when user explicitly requests QR code', async () => {
      await validateTelepathChanged('/')
    })

    it('keeps dialog open after the channel has changed and page rerendered', async () => {
      await validatePageRerendered('/')
    })
  })

  describe('when routing to contracts page', () => {
    it('renders correct page', async () => {
      await validateCorrectPageRendered('/contracts', 'Current value is:')
    })

    it('creates new telepath channel when user explicitly requests QR code', async () => {
      await validateTelepathChanged('/contracts')
    })

    it('keeps dialog open after the channel has changed and page rerendered', async () => {
      await validatePageRerendered('/contracts')
    })

    it('rerenders the page when channel changes', async () => {
      const { getByText, queryByText, store } = render(
        inRouter(Main, '/contracts')
      )
      const showQRCodeButton = await waitForElement(() =>
        getByText(/show qr code/i)
      )
      const {
        channelId: initialId,
        channelKey: initialKey
      } = store.getState().userData
      fireEvent.click(showQRCodeButton)
      // By waiting for channel to change we can request the button *AFTER* the page
      // has been rerendered. If we do not do that, we will get the reference
      // to a wrong button, which will close the dialog by changing the redux state
      // (AppEventsActions.setDialogClosed()), but react will warn that
      // we try to update state on an unmounted component.
      await wait(() => {
        expect(store.getState().userData.channelId).not.toEqual(initialId)
        expect(store.getState().userData.channelKey).not.toEqual(initialKey)
      })
      const doneButton = getByText(/done/i)
      fireEvent.click(doneButton)
      expect(queryByText(/scan the qr code/i)).toBeNull()
    })
  })
})
