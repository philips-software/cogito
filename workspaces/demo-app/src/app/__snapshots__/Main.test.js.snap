// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main when routing to contracts page renders correct page 1`] = `
.css-0,
[data-css-0] {
  display: grid;
  grid-gap: 5px;
  grid-template-areas: "header         header" "demo           documentation" "demoNavigation documentation" "footer         footer";
  grid-template-columns: 1fr 1fr;
  grid-template-rows: max-content 1fr max-content max-content;
  box-sizing: border-box;
  height: 100vh;
  -ms-grid-gap: 5px;
  -ms-grid-template-areas: "header         header" "demo           documentation" "demoNavigation documentation" "footer         footer";
  -ms-grid-template-columns: 1fr 1fr;
  -ms-grid-template-rows: max-content 1fr max-content max-content;
}

.css-1,
[data-css-1] {
  grid-area: header;
  padding: 15px;
  -ms-grid-area: header;
}

.css-2,
[data-css-2] {
  width: 100%;
  height: 100px;
  background-color: black;
  padding: 10px;
}

.css-3,
[data-css-3] {
  grid-area: demo;
  padding: 15px;
  -ms-grid-area: demo;
}

.css-4,
[data-css-4] {
  grid-area: demoNavigation;
  padding: 15px;
  -ms-grid-area: demoNavigation;
}

.css-5,
[data-css-5] {
  grid-area: documentation;
  padding: 15px;
  border-left: 1px solid black;
  overflow: auto;
  -ms-grid-area: documentation;
}

.css-6,
[data-css-6] {
  grid-area: footer;
  padding: 15px;
  -ms-grid-area: footer;
}

.css-7,
[data-css-7] {
  justify-content: flex-end;
  height: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
}

.css-8,
[data-css-8] {
  color: #ff80c3;
  font-size: 0.8rem;
}

<div>
  <div
    class="css-0"
  >
    <div
      class="css-1"
    >
      <div
        class="css-2"
      >
        <div
          class="ui huge header"
          style="color: white;"
        >
          Cogito
        </div>
        <div
          class="ui medium header"
          style="color: white; margin-top: 0px;"
        >
          Executing Contracts
        </div>
      </div>
    </div>
    <div
      class="css-3"
    >
      <div
        class="emotion-6 emotion-7"
      >
        <div
          style="margin: 20px 0px 20px 0px; padding: 0px; box-sizing: border-box;"
        >
          <div
            class="emotion-6 emotion-7"
          >
            <p>
              Current value is:
            </p>
            <div
              class="emotion-2 emotion-3"
              data-testid="current-value"
            >
              0
            </div>
            <div
              class="emotion-4 emotion-5"
            >
              <button
                class="ui black secondary button"
                role="button"
              >
                Increase by 5...
              </button>
              <button
                class="ui black secondary button"
                role="button"
              >
                Show QR code
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-4"
    >
      <div
        class="emotion-6 emotion-7"
      >
        <div
          aria-expanded="false"
          class="ui button floating labeled upward top pointing dropdown icon"
          role="listbox"
          tabindex="0"
        >
          <div
            aria-live="polite"
            class="text"
            role="alert"
          >
            Switch demo...
          </div>
          <i
            aria-hidden="true"
            class="exchange icon"
          />
          <div
            class="menu transition"
          >
            <div
              aria-checked="false"
              class="item"
              role="option"
            >
              <span
                class="text"
              >
                Cogito Identity
              </span>
            </div>
            <div
              aria-checked="true"
              class="active item"
              role="option"
            >
              <span
                class="text"
              >
                Executing Contracts
              </span>
            </div>
            <div
              aria-checked="false"
              class="item"
              role="option"
            >
              <span
                class="text"
              >
                Cogito Attestations
              </span>
            </div>
            <div
              aria-checked="false"
              class="item"
              role="option"
            >
              <span
                class="text"
              >
                Simple Encryption
              </span>
            </div>
            <div
              aria-checked="false"
              class="item"
              role="option"
            >
              <span
                class="text"
              >
                Stream Encryption
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-5"
    >
      <div />
    </div>
    <div
      class="css-6"
    >
      <div
        class="css-7 emotion-6 emotion-7"
      >
        <p
          class="css-8"
        >
          Â© 2019 by Cogito
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Main when routing to home page can update identity when telepath changes 1`] = `<div />`;

exports[`Main when routing to home page renders correct page 1`] = `
.css-0,
[data-css-0] {
  display: grid;
  grid-gap: 5px;
  grid-template-areas: "header         header" "demo           documentation" "demoNavigation documentation" "footer         footer";
  grid-template-columns: 1fr 1fr;
  grid-template-rows: max-content 1fr max-content max-content;
  box-sizing: border-box;
  height: 100vh;
  -ms-grid-gap: 5px;
  -ms-grid-template-areas: "header         header" "demo           documentation" "demoNavigation documentation" "footer         footer";
  -ms-grid-template-columns: 1fr 1fr;
  -ms-grid-template-rows: max-content 1fr max-content max-content;
}

.css-1,
[data-css-1] {
  grid-area: header;
  padding: 15px;
  -ms-grid-area: header;
}

.css-2,
[data-css-2] {
  width: 100%;
  height: 100px;
  background-color: black;
  padding: 10px;
}

.css-3,
[data-css-3] {
  grid-area: demo;
  padding: 15px;
  -ms-grid-area: demo;
}

.css-4,
[data-css-4] {
  grid-area: demoNavigation;
  padding: 15px;
  -ms-grid-area: demoNavigation;
}

.css-5,
[data-css-5] {
  grid-area: documentation;
  padding: 15px;
  border-left: 1px solid black;
  overflow: auto;
  -ms-grid-area: documentation;
}

.css-6,
[data-css-6] {
  grid-area: footer;
  padding: 15px;
  -ms-grid-area: footer;
}

.css-7,
[data-css-7] {
  justify-content: flex-end;
  height: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
}

.css-8,
[data-css-8] {
  color: #ff80c3;
  font-size: 0.8rem;
}

<div>
  <div
    class="css-0"
  >
    <div
      class="css-1"
    >
      <div
        class="css-2"
      >
        <div
          class="ui huge header"
          style="color: white;"
        >
          Cogito
        </div>
        <div
          class="ui medium header"
          style="color: white; margin-top: 0px;"
        >
          Identity
        </div>
      </div>
    </div>
    <div
      class="css-3"
    >
      <div
        class="emotion-8 emotion-9"
      >
        <div
          style="margin: 20px 0px 20px 0px; padding: 0px; box-sizing: border-box;"
        >
          <div
            class="emotion-8 emotion-9"
          >
            <p>
              Your Cogito account address is:
            </p>
            <div
              class="emotion-2 emotion-3"
              data-testid="current-address"
            >
              unknown
            </div>
            <div
              style="margin: 20px 0px 0px 0px; padding: 0px; box-sizing: border-box;"
            />
            <p>
              You are known as:
            </p>
            <div
              class="emotion-2 emotion-3"
              data-testid="current-username"
            >
              unknown
            </div>
            <div
              class="emotion-6 emotion-7"
            >
              <button
                class="ui black secondary button"
                role="button"
              >
                Read your identity...
              </button>
              <button
                class="ui black secondary button"
                role="button"
              >
                Show QR code
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-4"
    >
      <div
        class="emotion-8 emotion-9"
      >
        <div
          aria-expanded="false"
          class="ui button floating labeled upward top pointing dropdown icon"
          role="listbox"
          tabindex="0"
        >
          <div
            aria-live="polite"
            class="text"
            role="alert"
          >
            Switch demo...
          </div>
          <i
            aria-hidden="true"
            class="exchange icon"
          />
          <div
            class="menu transition"
          >
            <div
              aria-checked="true"
              class="active item"
              role="option"
            >
              <span
                class="text"
              >
                Cogito Identity
              </span>
            </div>
            <div
              aria-checked="false"
              class="item"
              role="option"
            >
              <span
                class="text"
              >
                Executing Contracts
              </span>
            </div>
            <div
              aria-checked="false"
              class="item"
              role="option"
            >
              <span
                class="text"
              >
                Cogito Attestations
              </span>
            </div>
            <div
              aria-checked="false"
              class="item"
              role="option"
            >
              <span
                class="text"
              >
                Simple Encryption
              </span>
            </div>
            <div
              aria-checked="false"
              class="item"
              role="option"
            >
              <span
                class="text"
              >
                Stream Encryption
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-5"
    >
      <div />
    </div>
    <div
      class="css-6"
    >
      <div
        class="css-7 emotion-8 emotion-9"
      >
        <p
          class="css-8"
        >
          Â© 2019 by Cogito
        </p>
      </div>
    </div>
  </div>
</div>
`;
